pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -167,6 +171,9 @@ func (r *ReconcileBareMetalHost) Reconcile(request reconcile.Request) (reconcile
 		return reconcile.Result{}, nil
 	}
 
+	// Clear any error so we can recompute it
+	host.ClearError()
```

Comment:
I'm trying to work through the implications of ignoring the return code here.

Shouldn't we save the result of this and then check that on line 455 instead of checking the result of clearing it a second time?

Reply:
Hrm, yes, you're right. Updating...

Reply:
Ah, no, I remember why that is now. If we had an error and clear it at the top of the loop, then do not reset the error, we're going to end up saving the status somewhere else through the process because whatever failed before will have worked this time. So by the time we get to the bottom of the loop there won't be anything to clear and that entire block needs to come out.